Docker changes

dabates opened the pull request:
- Startup.cs : Added debug values for some Variables to make sure they are passing correctly
- Dockerfile: refactored the build and added some directory removals (I've found they cause the build to error)
- README.md : added info on docker usage
- docker-compose : removed old entries
- Refactored the docker container to run the services more effeciently
- Adds scripts to run the db migration and setup the database and give the root user access

dabates committed:
Docker Changes

Initial changes
- Refactor how the docker container is started
- working on getting the startup / setup scripts working

dabates committed:
Working on getting the docker to build and start correctly

Commit:
Changes to the Docker files and scripts.  It now brings both containers up.  Runs the DB Migrations and sets everything up

dabates committed:
Fixes to make Docker Work

- Startup.cs : Added debug values for some Variables to make sure they are passing correctly
- Dockerfile: refactored the build and added some directory removals (I've found they cause the build to error)
- README.md : added info on docker usage
- docker-compose : removed old entries

danielunderwood commented on Dockerfile:
```diff
@@ -1,12 +1,19 @@
+COPY docker-services/LidarrAPI/docker-entrypoint.sh ./
+
+# There are some extra directories that cause issues for a build if they exist, remove them
+RUN rm -fr -- bin
```
Might want to reconsider this since docker should start with a fresh image when building. If these directories are sticking around, it may be an indication of a different issue. If it does need to stay in, I think each `RUN` creates an intermediate container, so it may be better to do `RUN rm -rf bin obj Properties Debug`.

dabates replied:
If we are guarnteed that it will be a fresh instance each time.. I Think it's kosher to get rid of them..

danielunderwood commented on README.md:
```diff
@@ -6,3 +6,14 @@ This is the update API of [https://github.com/Lidarr/Lidarr](https://github.com/
+
+If you would like to use the docker setup we have for this project, follow these directions:
+- Setup Environment Variables
+	- In the `docker-services/mysql/Dockerfile` 
```
Could we specify this at runtime or does it have to be at build time?

danielunderwood commented on docker-compose.yml:
```diff
@@ -1,23 +1,21 @@
         environment:
             - DataDirectory=/data
-            - Database="server=127.0.0.1;user id=root;password=${MYSQL_ROOT_PASSWORD};database=lidarrupdate;CharSet=utf8mb4"
+            - Database=server=mysql;user id=root;password=root;database=lidarrupdate;CharSet=utf8mb4
```
Why take out root password by environment variable?

danielunderwood replied:
And can we verify that the `Database` variable in the container is what we expect? I think it may be fine, but I'm worried that the equals and semicolons could cause some issues with the quotes taken out.

dabates commented:
To answer all your questions.. all these things was my misunderstanding of Docker.. I have a better way to do most of this now.. I'll get a this updated as soon as I push

dabates committed:
Refactor

Refactored the docker-compose, as my in-experince with docker containers skewed how I did some things.. this shoule be more streamlined

dabates commented on docker-compose.yml:
```diff
@@ -1,23 +1,27 @@
+        ports:
+            ## Specify a port here so that you can connect to the mysql server from the host
+            - 33060:3306
 
```
Note: This port here is only if you want connect into your mysql from your HOST (windows / linux ) without `exec`ing into the container.. can be removed if you don't think it's necessary

danielunderwood replied:
I think it may be best to leave it out for the moment unless we need outside access to the container. We could always put it back in if we come across a need to access the db from outside.

dabates committed:
Refactor for using ENV Variables

This refactors the WebAPI Code so that it will use ENV variables if they are present

dabates committed:
fixup! 
removes mysql port

dabates committed:
Adds more debug info
